File: tests/python/band_power_all.py
import argparse
import time
import brainflow
import numpy as np

from brainflow.board_shim import BoardShim, BrainFlowInputParams, LogLevels, BoardIds
from brainflow.data_filter import DataFilter, FilterTypes, AggOperations, WindowFunctions, DetrendOperations


def main():
    BoardShim.enable_dev_board_logger()

    # use synthetic board for demo
    params = BrainFlowInputParams()
    board_id = BoardIds.SYNTHETIC_BOARD.value
    sampling_rate = BoardShim.get_sampling_rate(board_id)
    board = BoardShim(board_id, params)
    board.prepare_session()
    board.start_stream()
    BoardShim.log_message(LogLevels.LEVEL_INFO.value, 'start sleeping in the main thread')
    time.sleep(10)
    data = board.get_board_data()
    board.stop_stream()
    board.release_session()

    eeg_channels = BoardShim.get_eeg_channels(board_id)
    bands = DataFilter.get_avg_band_powers(data, eeg_channels, sampling_rate, True)
    print("avg band powers : %s" % str(bands[0]))
    print("stddev band powers : %s" % str(bands[1]))


if __name__ == "__main__":
    main()

File: swift_package/examples/tests/band_power_all/band_power_all.swift
import Foundation

@main
struct testBandPowerAll {
    static func main() throws {
        // use synthetic board for demo
        try BoardShim.enableDevBoardLogger()
        let params = BrainFlowInputParams()
        let boardId = BoardIds.SYNTHETIC_BOARD
        let boardDescription = try BoardShim.getBoardDescr(boardId)
        let samplingRate = boardDescription.sampling_rate 
        let board = try BoardShim(boardId, params)
        try board.prepareSession()
        try board.startStream()
        try BoardShim.logMessage(.LEVEL_INFO, "start sleeping in the main thread")
        sleep(5)
        let nfft = try DataFilter.getNearestPowerOfTwo(samplingRate)
        var data = try board.getBoardData()
        try board.stopStream()
        try board.releaseSession()
    
        let eegChannels = try BoardShim.getEEGchannels(boardId)
        let bands = try DataFilter.getAvgBandPowers(data: data, channels: eegChannels, samplingRate: samplingRate, applyFilter: true)
        print("avg band powers : \(bands.0)")
        print("stddev band powers : \(bands.1)")
    }
}
